api/client.go

```diff
@@ -132,7 +132,7 @@ func (c *Client) do(ctx context.Context, method, path string, reqData, respData
 const maxBufferSize = 512 * format.KiloByte
 
 func (c *Client) stream(ctx context.Context, method, path string, data any, fn func([]byte) error) error {
-	var buf *bytes.Buffer
```

Comment:
This would result in a nil pointer dereference in the case the body was not set (like in a GET request). Fixed by not using a nil pointer, the behavior is the same.

Reply:
Nothing dereferences `buf` so this is inconsequential. You're ultimately passing the same thing to `NewRequestWithContext` since an uninitialized interface is also `nil.` This change is a noop though I'm not against it for consistency with `client.do`

Reply:
It dereferences here outside our code:
```
request, err := http.NewRequestWithContext(ctx, method, requestURL.String(), buf)
```

The test caught the behavior.

Reply:
My bad, you're right. The value being passed into `http.NewRequestWithContext` is `(*bytes.Buffer)(nil)` not `nil` so the `nil` checking in that function is ineffective